dex/src/test/java/io/crate/testing/SingleColumnBatchIterator.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.testing;
+
+import io.crate.concurrent.CompletableFutures;
+import io.crate.data.BatchIterator;
+import io.crate.data.CloseAssertingBatchIterator;
+import io.crate.data.InputList;
+
+import java.util.Iterator;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.IntStream;
+import java.util.stream.LongStream;
+
+/**
+ * A batch iterator with a single column.
+ *
+ * @param <T> the type of the column values
+ */
+public class SingleColumnBatchIterator<T> implements BatchIterator {
```

Comment:
Could this be changed to a static factory style class using the IterableBackedBatchIterator instead of re-implementing it itself?